webchat/utils/chat.js

```diff
@@ -0,0 +1,207 @@
+'use strict';
+(function () {
+  const app = angular.module('webchat');
+
+  app.factory('Chat', ['ChatMessage', (ChatMessage) => {
+    const Chat = class Chat {
+      /**
+       * Creates a new Chat object, which will handle a webrtc peer connection,
+       * its events and video/audio/data channels functionality.
+       * @param {MediaStream} stream - video/audio stream object
+       * @param {string} selfId - firebase id of the current user
+       */
+      constructor(stream, selfId) {
+        this.selfId = selfId;
+        this._selfStream = stream;
+        this.rpc = new RTCPeerConnection([{
```

Comment:
this rtc conection can fail  ?

Reply:
if yes, handle it

Reply:
Not at creation time, no. This is just a constructor to a new RTCPeerConnection